src/test/java/se/sundsvall/document/integration/db/model/DocumentTypeIdTest.java

```diff
@@ -0,0 +1,58 @@
+package se.sundsvall.document.integration.db.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+class DocumentTypeIdTest {
+
+	private static final String MUNICIPALITY_ID = "municipalityId";
+	private static final String TYPE = "type";
+	private static final DocumentTypeId BEAN = new DocumentTypeId(MUNICIPALITY_ID, TYPE);
+
+	@Test
+	void testConstructor() {
+		assertThat(BEAN)
+			.isNotNull()
+			.hasNoNullFieldsOrProperties()
+			.hasFieldOrPropertyWithValue("municipalityId", MUNICIPALITY_ID)
+			.hasFieldOrPropertyWithValue("type", TYPE);
+	}
+
+	@Test
+	void testToString() {
+		assertThat(BEAN).hasToString("DocumentTypeId [municipalityId=municipalityId, type=type]");
+	}
+
+	@Test
+	void testEquals() {
+		final var beanWithSameContent = new DocumentTypeId(MUNICIPALITY_ID, TYPE);
+		final var beanWithOtherContent1 = new DocumentTypeId("other" + MUNICIPALITY_ID, TYPE);
+		final var beanWithOtherContent2 = new DocumentTypeId(MUNICIPALITY_ID, "other" + TYPE);
+		new DocumentTypeId("other" + MUNICIPALITY_ID, TYPE);
+
+		final var beanOfOtherClass = new Object();
+
+		assertThat(BEAN)
+			.isEqualTo(BEAN)
```

Comment:
```
	assertThat(BEAN)
			.isEqualTo(BEAN)
```
seems a bit redudant?

Reply:
Code tested is:
```
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (!(obj instanceof final DocumentTypeId other)) {
			return false;
		}
		return Objects.equals(municipalityId, other.municipalityId) && Objects.equals(type, other.type);
	}
```
The assertion will result in a failure if the first if-statement (`if (this == obj)`) is tampered with.

Reply:
The `if (this == obj)` check is essentially there for an optimization (quickly confirming that the same object is equal to itself). In practice, it's almost universally reliable because it’s fundamental behavior in Java’s `==` operator, so it would rarely need testing explicitly unless you're doing something exceptionally weird with identity comparison.

In other words, the redundancy in the assertions doesn't add much real-world value. Testing with `beanWithSameContent` covers your equals implementation’s logic. The `==` check is so basic that if it fails, there are deeper JVM issues at play.

Reply:
In other model classes we are using beanmatchers to verify that there is valid equals, hashcode and toString methods. Why are we explicitly testing for this class?